gr-blocks/include/gnuradio/blocks/selector.h

```diff
@@ -32,11 +32,25 @@ class BLOCKS_API selector : virtual public block
 public:
     typedef std::shared_ptr<selector> sptr;
 
-    static sptr
-    make(size_t itemsize, unsigned int input_index, unsigned int output_index);
+    /*!
+     * Create new selector block and return a shared pointer to it
+     *
+     * \param itemsize size of the input and output items
+     * \param input_index the initially active input index
+     * \param output_index the initally active output index
+     * \param sync_consume consume the exact same amount of items from all inputs (true)
```

Comment:
Does anyone actually need this new mode? Seems a bit unnecessary to me.

The old mode (I think) was trying to do what the new version, with `sync_mode == false` is doing, except it didn't account for the fact that the input/output items may be different in size.

And regardless of this change, you can't switch selection on a particular item index, either.

Reply:
> Does anyone actually need this new mode? Seems a bit unnecessary to me.

Imagine this: 4 Wavfile sources with different file lengths, hence potentially different "workload chunks", connected to one audio sink through `selector`. 

Do you want them to stay in sync? in that case, `sync_consume`. Do you just want to consume items as they are available from the inactive inputs, without rate-unlimiting them? `~sync_consume`

I do feel like `sync_consume` is the more "understandable" and useful mode, but it seems there's different perspectives on that!

(the oldold blks2 behaviour would be to connect all inactive inputs to null sinks, which is kind of the opposite of `~sync_consume` mode, but honestly, I'm having a bit of a harder time finding a good use case for *that*. Suddenly, throttles in flowgraphs that should have no throttles become a necessity.)

Reply:
I can actually see two different types of non `sync_consume` mode:
- Absolutely no rate limit on inactive inputs. So consume the entirety of the input buffer, as fast as it fills up. -> needs upstream throttling
- Inactive inputs samples are consumed _up to_ the output rate. If more comes up, buffer will fill up. If there is less, samples are consumed as they are available. -> downstream backpressure handles throttling